### PR TITLE
[Merged by Bors] - feat: supporting lemmas for defining root systems

### PR DESCRIPTION
**Pull request by ocfnash**
A collection of loosely-related lemmas, split out from other work in the hopes of simplifying review.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/LinearAlgebra/Finsupp.lean [hunk 1]
```diff
@@ -760,6 +760,18 @@ theorem mem_span_iff_total (s : Set M) (x : M) :
   (SetLike.ext_iff.1 <| span_eq_range_total _ _) x
 #align finsupp.mem_span_iff_total Finsupp.mem_span_iff_total
 
+@[simp]
+lemma span_setOf_mem_eq_top {s : Set M} :
+    span R {x : span R s | (x : M) ∈ s} = ⊤ := by
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Finsupp.lean:765]: This is defeq to `span_span_coe_preimage`.

Comment (jcommelin) [reply Mathlib/LinearAlgebra/Finsupp.lean:765]: But maybe it's good to keep this as simp lemma.

File: Mathlib/LinearAlgebra/Finsupp.lean [hunk 2]
```diff
@@ -760,6 +760,18 @@ theorem mem_span_iff_total (s : Set M) (x : M) :
...
+  refine Submodule.eq_top_iff'.mpr fun ⟨z, hz⟩ ↦ ?_
+  rw [Finsupp.mem_span_iff_total] at hz ⊢
+  obtain ⟨l, rfl⟩ := hz
+  let f : s ↪ {x : span R s | (x : M) ∈ s} :=
+    ⟨fun x ↦ ⟨Set.inclusion (Submodule.subset_span) x, x.property⟩, fun x y ↦ by simp⟩
+  use l.embDomain f
+  rw [Finsupp.total_embDomain, Subtype.ext_iff, ← Submodule.coeSubtype, Finsupp.apply_total]
+  congr
```
Comment (jcommelin) [inline Mathlib/LinearAlgebra/Finsupp.lean:773]:
```suggestion
  apply span_span_coe_preimage
```

**Review by jcommelin — APPROVED** [review]
bors d+

File: Mathlib/LinearAlgebra/Finsupp.lean (hunk 2 above)
Comment (ocfnash) [reply Mathlib/LinearAlgebra/Finsupp.lean]: Aha, thanks!

**Comment by ocfnash** [comment]
bors merge

**Comment by ocfnash** [comment]
Looks like bors hit some sporadic error, trying again.

bors merge

**Pull request closed without merging**